glean-core/ios/GleanTests/Metrics/EventMetricTests.swift

```diff
@@ -237,4 +237,62 @@ class EventMetricTypeTests: XCTestCase {
         XCTAssertNotNil(events)
         XCTAssertEqual(1, events?.count)
     }
+
+    func testFlushQueuedEventsOnStartupDroppingPreinitEvents() {
+        setupHttpResponseStub()
+        expectation = expectation(description: "Completed upload")
+
+        let event = EventMetricType<SomeExtraKeys>(
+            category: "telemetry",
+            name: "test_event",
+            sendInPings: ["events"],
+            lifetime: .ping,
+            disabled: false,
+            allowedExtraKeys: ["some_extra"]
+        )
+
+        event.record(extra: [.someExtra: "run1"])
+        XCTAssertEqual(1, try! event.testGetValue().count)
+
+        Dispatchers.shared.setTaskQueuing(enabled: true)
+        event.record(extra: [.someExtra: "pre-init"])
+
+        Glean.shared.resetGlean(clearStores: false)
+
+        event.record(extra: [.someExtra: "post-init"])
+
+        waitForExpectations(timeout: 5.0) { error in
+            XCTAssertNil(error, "Test timed out waiting for upload: \(error!)")
+        }
+
+        let events = lastPingJson?["events"] as? [Any]
+        XCTAssertNotNil(events)
+        XCTAssertEqual(1, events?.count)
+        XCTAssertEqual("run1", ((events?[0] as? [String: Any])?["extra"] as? [String: Any])?["some_extra"] as? String)
+
+        setupHttpResponseStub()
+        expectation = expectation(description: "Completed upload")
+
+        Glean.shared.sendPingsByName(pingNames: ["events"])
+
+        waitForExpectations(timeout: 5.0) { error in
+            XCTAssertNil(error, "Test timed out waiting for upload: \(error!)")
+        }
+
+        let events2 = lastPingJson?["events"] as? [Any]
+        XCTAssertNotNil(events2)
+        XCTAssertEqual(2, events2?.count)
+        XCTAssertEqual(
+            "pre-init",
+            (
+                (events2?[0] as? [String: Any])?["extra"] as? [String: Any]
+            )?["some_extra"] as? String
+        )
```

Comment:
I feel like there's got to be a better way here that my novice swiftness just couldn't figure out...

Reply:
I think I would have created a helper function something like this:
```Swift
// Will throw if anything goes wrong
private func getExtraValue(from event: Any?, for key: String) -> String {
    let event = event! as! [String: Any]
    let extras = event["extra"] as! [String: Any]
    return extras[key] as! String
}
```

Then you could call it from the `XCTAssertEqual` tests like this:

```Swift
XCTAssertEqual("run1", getExtraValue(from: events![0], for: "some_extra"))
// or
XCTAssertEqual("pre-init", getExtraValue(from: events2![0], for: "some_extra"))
XCTAssertEqual("post-init", getExtraValue(from: events2![1], for: "some_extra"))
```

That at least puts most of the nasty force-unwrapping of optionals into a function.